### PR TITLE
LPS-188828 Method LayoutLocalServiceUtil.addLayout() of class LayoutLocalServiceUtil had it’s arguments changed

### PR DESCRIPTION
**Pull request by TamyrisTorres**
Ticket: https://liferay.atlassian.net/browse/LPS-188828

### CODE REVIEW

**Comment by TamyrisTorres** [comment]
ci:test:sf

**Comment by TamyrisTorres** [comment]
ci:test:relevant

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddLayoutParameterCheck.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+				continue;
+			}
+
+			String variableName = getVariableName(methodCall);
+
+			if (hasClassOrVariableName(
+					"LayoutLocalService", newContent, newContent, methodCall) ||
+				variableName.contains("LayoutLocalServiceUtil")) {
+
+				String newMethod = _toAssembleMethod(
```
Comment (natocesarrego) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddLayoutParameterCheck.java:61]: @TamyrisTorres, let's rename to `addMethodNewParameter`

File: modules/util/source-formatter/src/main/resources/sourcechecks.xml [hunk 2]
```diff
@@ -1598,6 +1598,10 @@
 			<category name="Upgrade" />
 			<description name="Fill the new parameter of the method `addFolder` of `JournalFolderService`, `JournalFolderLocalService`, and `JournalFolderLocalServiceUtil` classes" />
 		</check>
+		<check name="UpgradeJavaAddLayoutParameterCheck">
+			<category name="Upgrade" />
+			<description name="Fill the new parameter of the method `addLayout` of `LayoutLocalService`, and `LayoutLocalServiceUtil` classes" />
```
Comment (natocesarrego) [inline modules/util/source-formatter/src/main/resources/sourcechecks.xml:1603]: @TamyrisTorres, no comma needed after `LayoutLocalService` :wink:

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java [hunk 3]
```diff
@@ -637,7 +637,8 @@ protected String getVariableTypeName(
 	}
 
 	protected boolean hasClassOrVariableName(
-		String className, String content, String methodCall) {
+		String className, String content, String fileContent,
+		String methodCall) {
```
Comment (natocesarrego) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java:641]: @TamyrisTorres, should we just include this new parameter or does make sense we keep the method as it's, I mean, with 3 parameters, and create a new one? What do you think about it?

**Review by natocesarrego — CHANGES_REQUESTED** [review]

**Comment by TamyrisTorres** [comment]
ci:test:sf

**Comment by TamyrisTorres** [comment]
ci:test:relevant

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddLayoutParameterCheck.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		while (addLayoutMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, addLayoutMatcher.start());
+
+			List<String> parameterList = JavaSourceUtil.getParameterList(
+				methodCall);
+
+			if (parameterList.size() != 17) {
+				continue;
+			}
```
Comment (natocesarrego) [inline modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddLayoutParameterCheck.java:53]: All good with this pull request, @TamyrisTorres. I just have one last question to you: if tomorrow a new method with 17 parameters too is added to the classes `LayoutLocalService` and `LayoutLocalServiceUtil`, how this check will handle it? Will it be able to distinguish the 2 methods?

**Review by natocesarrego — APPROVED** [review]

**Comment by natocesarrego** [comment]
Good job, @TamyrisTorres! :smile:

**Review by natocesarrego — APPROVED** [review]

**Comment by natocesarrego** [comment]
Forwarded to Nícolas: https://github.com/nicolas-sss/liferay-portal/pull/47

**Pull request closed without merging**